civis/tests/test_io.py

```diff
@@ -152,6 +159,61 @@ def test_read_civis_pandas(self, *mocks):
                                  polling_interval=POLL_INTERVAL)
         assert df.equals(expected)
 
+    @pytest.mark.skipif(not has_pandas, reason="pandas not installed")
+    @pytest.mark.skipif(not has_numpy, reason="numpy not installed")
+    @mock.patch(api_import_str, return_value=civis_api_spec)
+    def test_read_civis_pandas_redshift_types(self, *mocks):
+        SQL_PANDAS_MAP = {
+            'smallint': np.int16,
+            'int': np.int32,
+            'integer': np.int32,
+            'bigint': np.int64,
+            'decimal': np.float64,
+            'numeric': np.float64,
+            'float': np.float64,
+            'float4': np.float32,
+            'float8': np.float64,
+            'real': np.float32,
+            'double precision': np.float64,
+            'boolean': np.bool,
+            'bool': np.bool,
+            'int2': np.int16,
+            'int4': np.int32,
+            'int8': np.int64,
+            'char': np.str,
+            'character': np.str,
+            'nchar': np.str,
+            'bpchar': np.str,
+            'varchar': np.str,
+            'character varying': np.str,
+            'nvarchar': np.str,
+            'text': np.str}
+
+        DATE_TYPES = ['date', 'timestamp', 'timestamptz',
+                      'timestamp without time zone']
+
+        # Construct all of these column types
+
+        pd_data = {}
+        for num, (name, type) in enumerate(SQL_PANDAS_MAP.items()):
+            if name in ['boolean', 'bool']:
+                value = [True]
+            elif name in ['char', 'character', 'nchar', 'bpchar', 'varchar',
+                          'character varying', 'nvarchar', 'text']:
+                value = ['foo']
+            elif name in DATE_TYPES:
+                value = ['05/07/1995']
+            else:
+                value = [num]
+            pd_data[name] = value
+
+        expected = pd.DataFrame
+
+        df = civis.io.read_civis('scratch.api_client_test_fixture',
+                                 'redshift-general', use_pandas=True,
+                                 dtype='redshift', polling_interval=POLL_INTERVAL)
+        assert df.equals(expected)
```

Comment:
Doing this test in the same way as `test_read_civis_pandas` would require recording a new VCR cassette.